src/StreamingDataFrames/streamingdataframes/models/serializers/quix.py

```diff
@@ -94,14 +177,14 @@ def __call__(self, value: bytes, ctx: SerializationContext) -> Iterable[dict]:
         :return: Iterable of dicts
         """
 
-        # Validate message headers
         # Warning: headers can have multiple values for the same key, but in context
         # of this function it doesn't matter because we're looking for specific keys.
-        headers_dict = (
-            {key: value.decode() for key, value in ctx.headers}
-            if ctx.headers is not None
-            else {}
-        )
+        if ctx.headers is None:
```

Comment:
`ctx.headers` could be `not None`, but miss the required headers, and still be in "legacy" format.

If we want to parse legacy and header-based formats with the same class, let's make the check more robust and validate that at least one of the required headers is present before setting the `legacy = True` flag.

Reply:
I changed the approach to inspect the raw message and see if "C" and "K" exist at the beginning of the message